app/script/calling/VideoGridRepository.js

```diff
@@ -35,26 +35,42 @@ z.calling.VideoGridRepository = class VideoGridRepository {
     const selfStreamState = callingRepository.selfStreamState;
     const calls = callingRepository.calls;
     this.grid = ko.observableArray([0, 0, 0, 0]);
-    this.selfId = ko.observable();
     this.thumbnailStream = ko.observable();
-    this.mirrorSelf = ko.pureComputed(() => !selfStreamState.screenSend());
 
     const selfStream = ko.pureComputed(() => {
-      return selfStreamState.videoSend() || selfStreamState.screenSend() ? localMediaStream() : undefined;
+      const stream = selfStreamState.videoSend() || selfStreamState.screenSend() ? localMediaStream() : undefined;
+      return {
+        id: stream && stream.id,
+        isMirrored: !!selfStreamState.videoSend(),
+        isMuted: !selfStreamState.audioSend(),
+        isSelf: true,
+        stream: stream,
+      };
     });
-    this.selfStreamMuted = ko.pureComputed(() => !selfStreamState.audioSend());
 
-    this.streams = ko.pureComputed(() => {
-      const noVideoParticipantIds = calls()
+    const videoParticipants = ko.pureComputed(() => {
```

Comment:
Why creating a new observable for this? This is only used inside the `streams` observable, let's keep it this way and not publish it in the parent scope

Reply:
it's just there for historical reasons. will fix.